mobile/src/main/java/org/openhab/habdroid/model/OpenHABItem.java

```diff
@@ -148,11 +148,11 @@ public Float getStateAsFloat() {
 		Float result;
 		// For uninitialized/null state return zero
 		if (state == null) {
-			result = new Float(0);
+			result = 0f;
```

Comment:
This also has the benefit of resulting in `Float.valueOf(0f)` being generated by the compiler when autoboxing the native `float`, where the JavaDoc says "[valueOf] should generally be used in preference to the constructor Float(float), as this method is likely to yield significantly better space and time performance by caching frequently requested values.".